### DISCUSSION TITLE
Merge and Knit

### DISCUSSION THREAD

**Discussion by rgaiacs** [quarto-dev/quarto-cli#1045] [Q&A] [answered]
I use bookdown and the “Merge and Knit” rendering approach described at https://bookdown.org/yihui/bookdown/new-session.html. My understanding is that Quarto only support the “Knit and Merge” rendering approach. Do you have any plan to support the “Merge and Knit” rendering approach?

**Comment by cderv** [comment, marked as answer]
For reference, related discussion https://github.com/quarto-dev/quarto-cli/discussions/431

"Merge and Knit" approach requires a single render for the entire book (one HTML file), then splitting the chapters into several pieces (as **bookdown** does). It is different from the current assumption of rendering independently each chapters (which bring us some more flexibility and feature that **bookdown** does not have). 

From  https://github.com/quarto-dev/quarto-cli/discussions/431#discussioncomment-2388170
> We don't currently support this model unfortunately (it's a non-trivial amount of work so not something we can knock out quickly)

Thanks for the suggestion though. 

I think in the short term we really need to document better how to render a book by sharing some data between chapters (which the Knit and Merge approach is really efficient for), and also leveraging [Projects Scripts](https://quarto.org/docs/projects/quarto-projects.html#project-scripts)

**Reply by cderv** [reply]
Related topics https://github.com/quarto-dev/quarto-cli/discussions/3577

**Reply by willgryan** [reply]
FWIW, here's how I share data between chapters with minimal work: https://gist.github.com/willgryan/6c8bb8fe1a394917d4779d2ca7ef09e7

I would love for a more Quarto-official way to do this

**Reply by cderv** [reply]
That is a really clever way to do this ! 

I don't know how / if we could make it a more official way as it would need to be something that work across engine. But opting in by using your solution is something interesting ! 

Maybe it could live in quarto R package  (github.com/quarto-dev/quarto-r) as this is specific to R users 🤔 ? 

Something to think about
